### PR TITLE
Move to vscode-languageclient 8.0.1

### LINKED ISSUE #711: Move to vscode-languageclient 8.0.1
With vscode-languageclient 8.0.1, we will benefit with:

 * LSP 3.7 support
 * we could remove shouldLanguageServerExitOnShutdown

### PR DESCRIPTION
**Pull request by angelozerr**
Move to vscode-languageclient 8.0.1

Fixes #711

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: src/client/clientErrorHandler.ts [hunk 1]
```diff
@@ -28,11 +28,19 @@ export class ClientErrorHandler implements ErrorHandler {
     this.heapDumpFolder = getHeapDumpFolderFromSettings() || context.globalStorageUri.fsPath;
   }
 
-  error(_error: Error, _message: Message, _count: number): ErrorAction {
-    return ErrorAction.Continue;
+  error(error: Error, message: Message, count: number): ErrorHandlerResult {
```
Comment (datho7561) [inline src/client/clientErrorHandler.ts:31]: Prefix the parameter names with `_` to indicate they aren't used

**Review by datho7561 — CHANGES_REQUESTED** [review]
One small change

File: src/client/clientErrorHandler.ts (hunk 1 above)
Comment (angelozerr) [reply src/client/clientErrorHandler.ts]: I didn't know this convention. Thanks for sharing that. fixed.

**Review by datho7561 — APPROVED** [review]
Works great! Thanks Angelo!

File: package-lock.json [hunk 2]
```diff
@@ -325,7 +325,7 @@
     },
     "@types/yauzl": {
       "version": "2.9.1",
-      "resolved": "https://registry.npmjs.org/@types/yauzl/-/yauzl-2.9.1.tgz",
+      "resolved": "https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/@types/yauzl/-/yauzl-2.9.1.tgz",
```
[the remark this answers is not in the document]
Comment (datho7561) [reply package-lock.json:328]: Oops yep my bad should have caught that

**Pull request merged**